File: backend/app/schemas/notification.py
from typing import Literal

from pydantic import BaseModel

Severity = Literal["info", "warning", "danger"]


class Notification(BaseModel):
    """Aviso operativo que la campanita muestra al usuario."""

    code: str
    module: str
    title: str
    description: str
    count: int
    severity: Severity


class NotificationFeed(BaseModel):
    """Avisos vigentes para el perfil que consulta.

    `total` es la suma de los pendientes, no la cantidad de avisos: es la cifra
    que se pinta sobre la campanita.
    """

    total: int
    items: list[Notification]

File: frontend/src/types/notification.ts
export type NotificationSeverity = 'info' | 'warning' | 'danger'

/** Aviso operativo calculado por el servidor sobre el estado real del sistema. */
export interface AppNotification {
  code: string
  /** Módulo al que pertenece; la campanita lo usa para saber a dónde llevar. */
  module: string
  title: string
  description: string
  count: number
  severity: NotificationSeverity
}

export interface NotificationFeed {
  /** Suma de pendientes: la cifra que se pinta sobre la campanita. */
  total: number
  items: AppNotification[]
}
